add logistic regression benchmark based off https://github.com/szilar…

cin opened the pull request:
added [logistic regression](https://github.com/szilard/benchm-ml/blob/master/1-linear/5-spark.txt) workload
changed partition and sleep workload to output expected values (for testing)
why am i ocd about dependency ordering?
lr conf example..should be expanded.

Commit:
add logistic regression benchmark based off https://github.com/szilard/benchm-ml/blob/master/1-linear/5-spark.txt
changed partition and sleep workload to output expected values (for testing)
why am i ocd about dependency ordering?
lr conf example..should be expanded.

ecurtin commented on lr-example.conf:
```diff
@@ -0,0 +1,26 @@
+
+  spark-contexts = [{
+
+    master = "spark://craigs-mbp:7077"
```
`craigs-mbp` should be localhost or change the whole string to `local[*]` or something.

ecurtin commented on lr-example.conf:
```diff
@@ -0,0 +1,26 @@
+        workloads = [
+          {
+            name = ["lr-bml"]
+            input = ["hdfs://craigs-mbp:9000/benchm-ml"]
```
See comment above

ecurtin commented on workloads/src/main/scala/com/ibm/sparktc/sparkbench/workload/ml/LogisticRegressionBenchmML.scala:
```diff
@@ -0,0 +1,104 @@
+// https://github.com/szilard/benchm-ml/blob/master/1-linear/5-spark.txt
+// ¯\_(ツ)_/¯
+
+case class LogisticRegressionBenchmML(
```
Nitpick: I might have gone with just LogisticRegression or LogisticRegressionWorkload

ecurtin commented on workloads/src/main/scala/com/ibm/sparktc/sparkbench/workload/ml/LogisticRegressionBenchmML.scala:
```diff
@@ -0,0 +1,104 @@
+          trainCount,
+          trainTime,
+          testFile,
+          trainTime,
```
Need to look at this logic closer for myself, but noting that I see trainTime twice in here

ecurtin commented on workloads/src/main/scala/com/ibm/sparktc/sparkbench/workload/ml/LogisticRegressionBenchmML.scala:
```diff
@@ -0,0 +1,104 @@
+// https://github.com/szilard/benchm-ml/blob/master/1-linear/5-spark.txt
+// ¯\_(ツ)_/¯
+
+case class LogisticRegressionBenchmML(
```
This is probably an offline discussion, but I'm interested to hear your reasons for using this implementation of LogisticRegression rather than the one in SparkTC/spark-bench. Simpler code? More flexibility?

cin replied:
I mainly chose it bc it uses a well-known, real-world dataset. All of the benchmarks look pretty straight forward too.

Commit:
udpates based on Emily's comments

Commit:
added unit tests for logistic regression workload
not totally happy with the way the input cfg property has to be set but not sure how to get around it

ecurtin committed:
Using SparkSessionProvider instead of DataFrameSuiteBase

ecurtin commented:
LGTM, merging to multiple-submit and then to develop.
